internal/yggdrasil/yggdrasil.go

```diff
@@ -314,6 +315,12 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 		if err != nil {
 			return operations.NewPostDataMessageForDeviceInternalServerError()
 		}
+		if heartbeat.Hardware != nil {
+			err = h.deviceRepository.UpdateLabels(ctx, edgeDevice, hardware.MapLabels(heartbeat.Hardware))
```

Comment:
Please make sure you've got the most recent changes and move that call to https://github.com/jakub-dzon/k4e-operator/blob/main/internal/heartbeat/synchronous-handler.go